code/modules/shuttle/dock_generator.dm

```diff
@@ -0,0 +1,120 @@
+/obj/item/whiteship_port_generator
+	name = "docking area signaller"
+	desc = "A signaling device for the NT expeditionary vessel, used to assign new docking areas."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-white"
+
+	var/list/placed_docks = list()
+	var/max_docks = 3
+
+/obj/item/whiteship_port_generator/examine(mob/user)
+	. = ..()
+	var/plural = length(placed_docks) > 1
+	. += "There [plural ? "are" : "is"] [max_docks - length(placed_docks)] use[plural ? "s" : ""] left."
+
+/obj/item/whiteship_port_generator/attack_self(mob/living/user)
+	if(is_station_level(user.z))
+		log_admin("[key_name(user)] attempted to create a whiteship dock in the station's sector at [COORD(user)].")
```

Comment:
is logging stuff that isn't/shouldn't be™️possible really needed?

Reply:
Yeah, we still want to see if people try to do griefy shit, even if it doesn't succeed.